package/kedro_viz/server.py

```diff
@@ -539,8 +544,18 @@ def nodes_metadata(node_id):
         dataset_metadata = _get_dataset_metadata(node)
         return jsonify(dataset_metadata)
 
-    # return empty JSON for parameters type
-    return jsonify({})
+    parameters = node["obj"]
+    if isinstance(parameters, dict):
```

Comment:
I'm not a big fan of such `isinstance` here. Can we maybe explicitly save parameter name in the dict we put in `_JSON_NODES`?
Something like:
```python
if "parameter_name" in node:
    # handle "params:..."
else:
    # handle "parameters"
```
@limdauto wdyt?

Reply:
Yeah, I can do that but how do you get "parameter_name" in this function? we can only access to `parameter_name` from `next(iter(parameters.items()))` correct?

Reply:
I meant I don't particularly like the fact that instead of the object we put the dictionary with name and object. Parameter name can have its own top-level key rather than being squeezed into the `obj` key.

Reply:
@DmitriiDeriabinQB Hope I understand you correctly, but it sounds like you're asking for the single-parameter endpoint to have a different structure from the multiple-parameter endpoint, is that right? Kiyo had it like that originally, but [I asked him to change it](https://github.com/quantumblacklabs/kedro-viz/pull/275#issuecomment-705170566).

If you think about this from the front-end perspective, we might want to separate out different keys to display them differently in the viz meta panel when clicking on a 'Parameters' node - e.g.:
![image](https://user-images.githubusercontent.com/1155816/95767286-70b8ce80-0cac-11eb-8ca7-407fdfa353c1.png)
but if you click on a single node, we might want to display it the same way, but if it's not in the same format we might not be able to recognise it as a distinct type just from the structure, because a parameter can be one of lots of different types. So we're using the same structure to keep it simple on the front end. Does that help? Hope I haven't misunderstood you

Reply:
@richardwestenra I'm still keeping the format you asked, his suggestion was more about how to generate it in code :)

Reply:
@richardwestenra yeah, I read through your discussion with Kiyo above and understand the solution. I wasn't talking about the format it is exposed to the frontend but rather the way the data stored in the backend. I'm moderately cautious about the complexity of the data wrangling logic on Python side (while the data itself is very trivial).
TLDR: It's purely the discussion about the Python side of things and should not affect the API contract.